libs/s25main/mapGenerator/HeadQuarters.h

```diff
@@ -42,45 +44,66 @@ namespace rttr { namespace mapGenerator {
      * @param area area within the HQ position should be
+     * @param distance desired player distance to mountain
      *
      * @return all suitable HQ positions witihn the specified area (or the entire map if the area is empty).
      * @throw runtime_error
      */
     template<class T_Container>
-    std::vector<MapPoint> FindHqPositions(const Map& map, const T_Container& area)
+    std::vector<MapPoint> FindHqPositions(const Map& map, const T_Container& area, MountainDistance distance)
     {
         if(area.empty())
         {
             throw std::runtime_error("Could not find any valid HQ position!");
         }
 
-        const auto isValid = [](const MapPoint& pt) { return pt.isValid(); };
-        const auto isObstacle = [&map](MapPoint point) {
-            return map.textureMap.Any(point, [](auto t) { return !t.Is(ETerrain::Buildable); });
+        const auto isObstacle = [&map](const MapPoint& pt) {
+            return map.textureMap.Any(pt, [](auto t) { return !t.Is(ETerrain::Buildable); });
         };
-        std::vector<MapPoint> hqs;
-        std::copy_if(map.hqPositions.begin(), map.hqPositions.end(), std::back_inserter(hqs), isValid);
 
-        // Quality of a map point as one player's HQ is a mix of:
-        // 1. distance to other players' HQs (higher = better)
-        // 2. distance to any other obstacle e.g. mountain & water (higher = better)
-        NodeMapBase<unsigned> potentialHqQuality = DistancesTo(hqs, map.size);
+        // Ensure possible HQ positions are far enough from any obstacles
         const auto obstacleDistance = DistancesTo(map.size, isObstacle);
-        const auto minDistance = helpers::clamp(GetMaximum(obstacleDistance, area), 2u, 4u);
+        const auto minObstacleDistance = helpers::clamp(GetMaximum(obstacleDistance, area), 2u, 4u);
+        const auto farFromObstacles = [&obstacleDistance, minObstacleDistance](const MapPoint& pt) {
+            return obstacleDistance[pt] >= minObstacleDistance;
+        };
+        std::vector<MapPoint> possiblePositions;
+        std::copy_if(area.begin(), area.end(), std::back_inserter(possiblePositions), farFromObstacles);
 
+        if(possiblePositions.empty())
+        {
+            return possiblePositions;
+        }
+
+        // Keep minimum distance to other HQs and desired distance to mountains
+        const auto mountain = [&map](const MapPoint& pt) { return map.textureMap.Any(pt, IsMinableMountain); };
+        const auto mountainDistances = DistancesTo(map.size, mountain);
+        const auto desiredDistance = static_cast<unsigned>(distance);
+        const auto distanceToOtherHqs = DistancesTo(map.hqPositions, map.size);
+        const unsigned minimumHqDistance = (map.size.x + map.size.y) / 16;
+        const int maximumRadius = (map.size.x + map.size.y) / 4;
         std::vector<MapPoint> positions;
-        for(const MapPoint& pt : area)
+        for(int radius = 1; positions.empty() && radius < maximumRadius; radius *= 2)
```

Comment:
So I changed the logic here to widen the radius to look for points with the desired mountain distance. 
I increment the radius by `*2` to avoid potential performance issues on larger maps.